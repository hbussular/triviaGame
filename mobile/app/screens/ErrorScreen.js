import React from 'react'
import { Image, Text, View } from 'react-native'
import Expo from 'expo'

import ScreenProvider from '../containers/ScreenProvider'
import PrimaryButton from '../components/PrimaryButton'

import styles from './ErrorScreen.style'
import warning from '../assets/images/warning.png'

const ErrorScreen = () => (
  <ScreenProvider>
    <View style={styles.viewContainer}>
      <Image
        style={styles.warningContainer}
        source={warning}
        resizeMode="contain"
      />
      <Text style={styles.textContainer}>
        Ooh, something went wrong!
      </Text>
      <View style={styles.bottomContainer}>
        <PrimaryButton
          label="Restart"
          onPress={() => Expo.Util.reload()}
        />
      </View>
    </View>
  </ScreenProvider>
)

export default ErrorScreen
