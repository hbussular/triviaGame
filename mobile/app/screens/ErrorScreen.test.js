import React from 'react'
import renderer from 'react-test-renderer'
import ErrorScreen from './ErrorScreen'

import { screenScenario } from '../tests/scenarios'

test('renders correctly', () => {
  const tree = renderer
    .create(<ErrorScreen {...screenScenario} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
