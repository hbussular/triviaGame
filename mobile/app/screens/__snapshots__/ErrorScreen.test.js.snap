// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F95E42",
      "height": "100%",
      "padding": 20,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "center",
      }
    }
  >
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": "30%",
          "width": "30%",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "rgba(255, 255, 255, 1)",
          "fontSize": 28,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Ooh, Sorry! Something happens!
    </Text>
    <View
      style={
        Object {
          "bottom": 0,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "position": "absolute",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 1)",
            "borderRadius": 8,
            "display": "flex",
            "opacity": 1,
            "paddingVertical": 20,
            "width": "100%",
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontWeight": "700",
                "textAlign": "center",
              },
              Object {
                "color": "rgba(213, 55, 43, 1)",
              },
              Object {
                "fontSize": 16,
              },
            ]
          }
        >
          Restart
        </Text>
      </View>
    </View>
  </View>
</View>
`;
